dot/rpc/subscription/helpers_test.go

```diff
@@ -30,17 +33,24 @@ func setupWSConn(t *testing.T) (*WSConn, *websocket.Conn, func()) {
 		}
 
 		wskt.Wsconn = c
+
+		// Notify that the WebSocket connection is ready
+		close(connReady)
 	}
 
 	server := httptest.NewServer(http.HandlerFunc(h))
 	defer server.Close()
 
 	wsURL := "ws" + strings.TrimPrefix(server.URL, "http")
 	ws, r, err := websocket.DefaultDialer.Dial(wsURL, nil)
-	defer r.Body.Close()
```

Comment:
is there any reason for removing the defer?

Reply:
because `r` is not being used and if I ignore it gives a lint error to close it.
so I think it's better to keep the `r` and close it immediately.

Reply:
You only need to close the body if you open it to unmarshal, in this case if `r` is not being used then there is no reason to close the body, I would suggest to ignore the `r`

```go
ws, _, err := websocket.DefaultDialer.Dial(wsURL, nil)
```

Reply:
Oh, okay...